tests/test_cookies.py

```diff
@@ -0,0 +1,501 @@
+    """ Test the attribute 'cookies' in Response object."""
+    http = httplib2.Http()
+
+    with tests.server_const_http(headers={"set-cookie": cookie_header}) as uri:
+        response, _ = http.request(uri, "GET")
+        assert len(response.cookies) == 3
+        assert response.cookies["A"] == "1"
+        assert response.cookies["B"] == "2"
+        assert response.cookies["C"] == "3"
+
+
+def test_onetime_cookie():
+    http = httplib2.Http()
+    cookies = {"AAA": "111", "BBB": "222"}
+    handler = tests.http_reflect_with_cookies()
+
+    with tests.server_request(handler) as uri:
+        response, content = http.request(uri, "GET", body="this request have one time cookie", cookies=cookies)
+        assert response.status == 200
+        assert dict([ck.split("=") for ck in content.decode().split("; ")]) == cookies
+
+
+@pytest.mark.parametrize("cookie_header", SET_COOKIE_HEADERS, ids=["single", "multiple"])
+def test_cookie_manager_enabled(cookie_header, cookie_http):
+    print(cookie_header)
+    handler = tests.http_reflect_with_cookies({"/login": cookie_header})
+
+    with tests.server_request(handler, request_count=2) as uri:
+        login_url = urllib.parse.urljoin(uri, "/login")
+        response, content = cookie_http.request(login_url, "POST", body="this is a login request")
+        assert response.status == 200
+        assert len(response.cookies) == 3
+
+        op_url = urllib.parse.urljoin(uri, "/op")
+        response, content = cookie_http.request(op_url, "POST", body="this is an op request")
+        assert response.status == 200
+
+        if isinstance(cookie_header, (list, tuple)):
+            cookie_header = ", ".join(cookie_header)
+        set_cookies = {ck["name"]: ck["value"] for ck in httplib2.CookieJar.parse_iter(cookie_header)}
+        assert dict([ck.split("=") for ck in content.decode().split("; ")]) == set_cookies
+
+
+def test_cookie_manager_expire(cookie_http):
+    expires = (datetime.utcnow() + timedelta(seconds=10)).strftime("%a, %d %b %Y %H:%M:%S GMT")
+    cookie_header = ("A=1", "B=2; Max-Age=5", "C=3; Expires={}".format(expires))
+    handler = tests.http_reflect_with_cookies({"/login": cookie_header})
+
+    with tests.server_request(handler, request_count=4, timeout=20) as uri:
+        login_url = urllib.parse.urljoin(uri, "/login")
+        response, content = cookie_http.request(login_url, "POST", body="this is a login request")
+        assert response.status == 200
+
+        op_url = urllib.parse.urljoin(uri, "/op")
+        response, content = cookie_http.request(op_url, "POST", body="op request 1")
+        assert response.status == 200
+
+        time.sleep(5)
+        response, content = cookie_http.request(op_url, "POST", body="op reques 2")
+        assert response.status == 200
+        set_cookies = {"A": "1", "C": "3"}
+        assert dict([ck.split("=") for ck in content.decode().split("; ")]) == set_cookies
+
+        time.sleep(5)
```

Comment:
```suggestion
        time.sleep(1)  # FIXME patch time
```

Reply:
Refer to https://github.com/httplib2/httplib2/pull/180#discussion_r527097967